platform/platform-logging/src/main/java/org/codice/ddf/logging/platform/LoggingServiceBean.java

```diff
@@ -0,0 +1,200 @@
+
+    private final ReadLock readLock = new ReentrantReadWriteLock().readLock();
+
+    private EvictingQueue<LogEvent> logEvents;
+
+    private int maxLogEvents = DEFAULT_LOG_EVENTS_LIMIT;
+
+    private ObjectName objectName;
+
+    private MBeanServer mBeanServer;
+
+    public LoggingServiceBean() {
+        try {
+            logEvents = EvictingQueue.create(DEFAULT_LOG_EVENTS_LIMIT);
+            objectName = new ObjectName(MBEAN_NAME);
+            mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        } catch (MalformedObjectNameException e) {
+            LOGGER.error("Unable to create Logging Service MBean with name [{}].", MBEAN_NAME, e);
+        }
+    }
+
+    public void init() {
+        try {
+            try {
+                mBeanServer.registerMBean(this, objectName);
+                LOGGER.info("Registered Logging Service MBean under object name: {}",
+                        objectName.toString());
+            } catch (InstanceAlreadyExistsException e) {
+                mBeanServer.unregisterMBean(objectName);
+                mBeanServer.registerMBean(this, objectName);
+                LOGGER.info("Re-registered Logging Service MBean");
+            }
+        } catch (MBeanRegistrationException | InstanceNotFoundException
+                | InstanceAlreadyExistsException | NotCompliantMBeanException e) {
+            LOGGER.error("Could not register MBean [{}].", objectName.toString(), e);
+        }
+    }
+
+    public void destroy() {
+        try {
+            if (objectName != null && mBeanServer != null) {
+                mBeanServer.unregisterMBean(objectName);
+                LOGGER.info("Unregistered Logging Service MBean");
+            }
+        } catch (InstanceNotFoundException | MBeanRegistrationException e) {
+            LOGGER.error("Exception unregistering MBean [{}].", objectName.toString(), e);
+        }
+    }
+
+    /**
+     * Called each time a {@link org.ops4j.pax.logging.spi.PaxLoggingEvent} is created in the system
+     */
+    @Override
+    public void doAppend(PaxLoggingEvent paxLoggingEvent) {
+        LogEvent logEvent = createLogEvent(paxLoggingEvent);
+        add(logEvent);
+    }
+
+    @Override
+    public List<LogEvent> retrieveLogEvents() {
+        try {
+            readLock.lock();
+            return Lists.newArrayList(logEvents);
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    /**
+     * Sets the maximum number of {@link LogEvent}s to store
+     * 
+     * @param newMaxLogEvents
+     *            This number cannot be less than 0 or greater than {@code MAX_LOG_EVENTS_LIMIT}. In
+     *            the event that this parameter is less than 0 or greater than
+     *            {@code MAX_LOG_EVENTS_LIMIT}, the maximum log events stored will set set to
+     *            {@code MAX_LOG_EVENTS_LIMIT}.
+     */
+    public void setMaxLogEvents(int newMaxLogEvents) {
+        try {
```

Comment:
Need to do arg check here and throw an exception if it's wrong. And don't forget the unit test ;-)